react-validated-input/my-app/src/ValidatedInput.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import { FaCheck, FaXmark } from 'react-icons/fa6';
+import './ValidatedInput.css';
+
+export function ValidatedInput() {
+  const [password, setPassword] = useState('');
+
+  const specialChars = [
+    '!',
+    '@',
+    '#',
+    '$',
+    '%',
+    '^',
+    '&',
+    '*',
+    '(',
+    ')',
+    ';',
+    '-',
+    '_',
+    '+',
+    '=',
+    '/',
+    '?',
+    '<',
+    '>',
+    '[',
+    ']',
+    '{',
+    '}',
+    '\\',
+    '|',
+  ];
+
+  const errorArr: string[] = [];
+  if (password.length === 0) {
+    errorArr.push('no password');
+  } else if (password.length < 8) {
+    errorArr.push('too short');
+  }
+
+  let hasNoNums = true;
```

Comment:
These checks are easier with regular expressions, but what you've done is probably the best you can do if you don't want to use regular expressions.